src/main/scala/fairysf/format/Forsyth.scala

```diff
@@ -17,62 +17,89 @@ object Forsyth {
+      .flatMap(uciMove => baseSituation.move(uciMove).toOption)
+      .map(_.situationAfter)
+      .getOrElse(baseSituation)
+    Some(finalSituation)
+
   }
 
   def <<(fen: FEN): Option[Situation] = <<@(Variant.default, fen)
 
-  case class SituationPlus(situation: Situation, fullTurnCount: Int) {
+  case class SituationPlus(situation: Situation, fullTurnCount: Int, halfTurnMarker: Boolean = false) {
 
     def turnCount = fullTurnCount * 2 - situation.player.fold(2, 1)
     // TODO: Set this for Amazons. Needs to have the prev move encoded into the FEN
     // to calculate this. This work has been done here:
     // https://github.com/Mind-Sports-Games/strategygames/compare/master...392-support-amazons-in-analysis-page#diff-b0b2b8a0e93f9c1f675971f22eda79cc6cfedee6f2284b31a0fba54832966933R178
     // but hasn't been merged yet
-    def plies     = turnCount
-
+    def plies     = turnCount * 2 + (if (halfTurnMarker) 1 else 0)
   }
 
-  def <<<@(variant: Variant, fen: FEN): Option[SituationPlus] =
+  def <<<@(variant: Variant, fen: FEN): Option[SituationPlus] = {
     <<@(variant, fen) map { sit =>
-      SituationPlus(
-        // not doing half move clock history like we do in chess
-        sit,
-        fen.value.split(' ').last.toIntOption.map(_ max 1 min 500) | 1
-      )
+      {
+        val splitted       = fen.value.split(' ')
+        val halfTurnMarker = splitted.lift(6).filter(_.startsWith("½")).fold(false)(_ => true)
+        val fullTurnCount  = splitted.lift(5).flatMap(_.toIntOption).map(_ max 1 min 500) | 1
+        SituationPlus(
+          // not doing half move clock history like we do in chess
+          sit,
+          fullTurnCount,
+          halfTurnMarker
+        )
+      }
     }
+  }
 
   def <<<(fen: FEN): Option[SituationPlus] = <<<@(Variant.default, fen)
 
   def >>(situation: Situation): FEN = >>(SituationPlus(situation, 1))
 
   def >>(parsed: SituationPlus): FEN =
     parsed match {
-      case SituationPlus(situation, _) =>
+      case SituationPlus(situation, _, _) =>
         >>(Game(situation, plies = parsed.plies, turnCount = parsed.turnCount))
     }
 
-  def >>(game: Game): FEN = exportBoardFen(game.situation.board)
+  // The reason we let variants choose which board to use for the fen is because
+  // some variants (amazon, for example) need to use the previous fen for half-moves
+  // or fairy stockfish won't parse it properly.
+  def >>(game: Game): FEN =
+    game.situation.board.variant.paramsForFen(game) match {
+      case (board, Some(lastMove)) => exportBoardFenWithLastMove(board, lastMove)
+      case (board, _)              => exportBoardFen(board)
+    }
 
   def exportBoard(board: Board): String = exportBoardFen(board).value
 
-  def exportBoardFen(board: Board): FEN = board.variant.exportBoardFen(board)
+  def exportBoardFen(board: Board): FEN                             = board.variant.exportBoardFen(board)
+  def exportBoardFenWithLastMove(board: Board, lastMove: Move): FEN =
```

Comment:
yeah so `lastAction` is already on `board.history`

Reply:
Is last action what I actually want?

Reply:
Let's chat about this tomorrow. I think we can definitely add enough things to History to make this work, but I'm not sure that the current attributes are the correct attributes